### PR TITLE
ContextualMenu: Added the ability to create header items

### PR DESCRIPTION
**Pull request by joschect**
#### Pull request checklist
- [x] Include a change request file if publishing 
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [x] Documentation update

#### Description of changes

Added in the ability to give the contextualmenu headers. Now you pass if you pass in an item type of ContextualMenuItemType.Header it will create a header.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.scss [hunk 1]
```diff
@@ -66,6 +66,26 @@ $ContextualMenu-iconWidth: 14px;
   }
 }
 
+.ms-ContextualMenu-header {
+  @include focus-border();
+
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.scss]: Nit: Don't need spaces here and on 86.

File: packages/office-ui-fabric-react/src/demo/pages/ContextualMenuPage/ContextualMenuPage.tsx [hunk 2]
```diff
@@ -18,6 +19,7 @@ const ContextualMenuBasicExampleCode = require('./examples/ContextualMenu.Basic.
 const ContextualMenuCheckmarksExampleCode = require('./examples/ContextualMenu.Checkmarks.Example.tsx');
 const ContextualMenuDirectionalExampleCode = require('./examples/ContextualMenu.Directional.Example.tsx');
 const ContextualMenuCustomizationExampleCode = require('./examples/ContextualMenu.Customization.Example.tsx');
+// const ContextualMenuHeaderExampleCode = require('./examples/ContextualMenu.Header.Example.tsx');
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/demo/pages/ContextualMenuPage/ContextualMenuPage.tsx]: delete

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.scss [hunk 3]
```diff
@@ -66,6 +66,24 @@ $ContextualMenu-iconWidth: 14px;
...
+  color: $ms-color-themePrimary;
+  background: none;
+  border: none;
+  height: $ContextualMenu-itemHeight;
+  line-height: $ContextualMenu-itemHeight;
+  cursor: default;
+  padding: 0px 6px;
+  user-select: none;
+  -moz-user-select: none;
+  -webkit-user-select: none;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.scss]: you dont' need the vendor prefixes. Postcss will take care of that for you. just use `user-select`.

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.Props.ts [hunk 4]
```diff
@@ -6,7 +6,11 @@ import { IIconProps } from '../Icon/Icon.Props';
 import { IRectangle } from '../../common/IRectangle';
 import { IPoint } from '../../common/IPoint';
 export { DirectionalHint } from '../../common/DirectionalHint';
-
+export enum ContextualMenuItemType {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.Props.ts]: nit, add blank line before/after enum

File: common/changes/contextualmenu-add-headers_2017-01-06-19-47.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "Contextualmenu: Added header item so that the contextualmenu can now have headers",
```
Comment (dzearing) [inline common/changes/contextualmenu-add-headers_2017-01-06-19-47.json]: nit: ContextualMenu casing

**Pull request merged**